source/client-side-operations-timeout/client-side-operations-timeout.rst

```diff
@@ -0,0 +1,904 @@
+==============================
+Client Side Operations Timeout
+==============================
+
+:Title: Client Side Operations Timeout
+:Author: Divjot Arora
+:Spec Lead: Jeff Yemin
+:Approvers: Jeff Yemin, Clyde Bazile
+:Advisory Group: Bernie Hackett, Jason Carey, Will Shulman
+:Status: Accepted
+:Type: Standards
+:Minimum Server Version: 2.6
+:Last Modified: November 24, 2020
+:Version: 1.0.0
+
+.. contents::
+
+--------
+
+Abstract
+========
+
+This specification outlines a new ``timeoutMS`` option to govern the amount
+of time that a single operation can execute before control is returned to the
+user. This timeout applies to all of the work done to execute the operation,
+including but not limited to server selection, connection checkout, and
+server-side execution.
+
+META 
+====
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”,
+“SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this
+document are to be interpreted as described in \`RFC 2119
+<https://www.ietf.org/rfc/rfc2119.txt>`_.
+
+Specification
+=============
+
+Terms
+-----
+
+min(a, b)
+  Shorthand for "the minimum of a and b" where ``a`` and ``b`` are numeric
+  values. For any cases where 0 means "infinite" (e.g. `timeoutMS`_),
+  ``min(0, other)`` MUST evaluate to ``other``.
+
+MongoClient Configuration
+-------------------------
+
+This specification introduces a new configuration option and deprecates some
+existing options.
+
+timeoutMS
+~~~~~~~~~
+
+This 64-bit integer option specifies the per-operation timeout value in
+milliseconds. The default value is unset. Both unset and an explicit value of
+0 mean infinite, though some client-side timeouts like
+``serverSelectionTimeoutMS`` will still apply. Drivers MUST error if a
+negative value is specified. This value MUST be configurable at the level of
+a MongoClient, MongoDatabase, MongoCollection, or of a single operation.
+However, if the option is specified at any level, it cannot be later changed
+to unset. At each level, the value MUST be inherited from the previous level
+if it is not explicitly specified.
```

Comment:
For a collection, is the database or client the "previous level"? If the collection MUST inherit timeoutMS from the database how we would enforce that since a collection object doesn't necessarily depend on a database object.

Reply:
The previous level is the database.

> since a collection object doesn't necessarily depend on a database object

Can you explain what you mean here? My understanding was that a MongoCollection always belongs to a MongoDatabase, which belongs to a MongoClient.